add support for AKS Sandbox via vscode-sandbox example (#236)
kubernetes-sigs/agent-sandbox: examples/vscode-sandbox/README.md

Validate that the `Pod` with gVisor enabled is running:

```shell
kubectl wait --for=condition=Ready sandbox sandbox-example
kubectl get pods -o jsonpath=$'{range .items[*]}{.metadata.name}: {.spec.runtimeClassName}\n{end}'
```

### Harden Agent Sandbox isolation using Kata Containers and Hyper-V (Optional)

Similarly to the gVisor scenario, there are scenarios where you may want to provide
workload isolation for running untrusted workloads inside a sandbox.

[AKS Sandbox](https://learn.microsoft.com/en-us/azure/aks/use-pod-sandboxing)
provides a hardware virtualization layer between the host operating system and the
containerized workload via [CloudHypervisor](https://www.cloudhypervisor.org) and

> Hyper-V. There is no shared kernel between the host and the guest workload.

Author: I was trying to draw a distinction between gVisor and MSHV. The section previous to this section discusses gVisor and its approach to virtualization. 

I'll omit Hyper-V and only discuss MSHV + CloudHypervisor.

This example demonstrates how to use `Sandbox` along with CloudHypervisor in order

> to utilize the lifecycle features of `Sandbox` in addition with hypervisor 

Reviewer: ```suggestion
to utilize the lifecycle features of `Sandbox` alongside the hypervisor 
```

Super small nit, think this flows a little better?

isolation features of CloudHypervisor + Hyper-V.

#### Create a cluster with AKS Sandboxing enabled

The scripts directory contains shell scripts for provisioning, applying, and
cleaning up Azure Kubernetes cluster resources.

From the [./scripts](./scripts/) directory, execute the following.

```shell
./az-provision.sh
# either bake a new sandbox-router image or use the pre-baked ghcr.io/devigned/sandbox-router:latest
./apply.sh --sandbox-router-image ghcr.io/devigned/sandbox-router:latest
```

The `./apply.sh` script applies resources needed for ease of use, but at the heart
of it, it's simply the following:

```shell
kubectl apply -k ./overlays/kata-mshv
```

Which applies the overlay that includes the runtime class for Kata hypervisor

> isolation (MSHV: Microsoft Hypervisor).

Author: I would suggest specifying the backend as it would provide a differentiation from the kata + qemu example in the same parent directory.
